### DISCUSSION TITLE
Seeking Assistance with Error: HTTP/2 stream 1 was not closed cleanly: PROTOCOL_ERROR (err 1)

### DISCUSSION THREAD

**Discussion by realyukii** [curl/curl#13152] [Q&A]
Greetings fellow developers,

I hope this message finds you well.

I'm reaching out to the community seeking assistance with an error I've encountered while downloading a file from [this link](https://releases.wikimedia.org/mediawiki/1.41/mediawiki-1.41.0.tar.gz). While I may not fully grasp the intricacies of the technical details, I'd be grateful if someone could lend their expertise to help me understand the root cause of this issue and guide me towards a resolution.

Description of the Issue:
Steps to Reproduce: `curl -v -s -LOJ --trace-config all <url>  2>&1 | tee /tmp/curl-tar.log`
I've attached the log file for your reference: [curl-tar.log](https://github.com/curl/curl/files/14661497/curl-tar.log)
I've also tried:
- download it from the browser (finished without error, but cannot extract it with tar: `gzip: stdin: unexpected end of file`)
- downgrade curl to use http 1.1, but the issue persists. (maybe this proves the issue is on the server-side, not in curl)

What I'm Hoping For:
I would be incredibly appreciative if someone could:
1. Analyze the log file and provide insights into the underlying cause of the error.
2. Offer guidance on how to effectively resolve the issue.
3. Share any relevant knowledge or resources that could aid in my understanding.

I understand that debugging can be a complex process, and I sincerely value your time and expertise. Any assistance you can provide would be greatly welcomed.

Thank you in advance for your consideration and support.

**Comment by icing** [comment]
Hi,

> Am 20.03.2024 um 04:03 schrieb Yuki San ***@***.***>:
> 
> 
> Greetings fellow developers,
> I hope this message finds you well.
> I'm reaching out to the community seeking assistance with an error I've encountered while downloading a file from this link. While I may not fully grasp the intricacies of the technical details, I'd be grateful if someone could lend their expertise to help me understand the root cause of this issue and guide me towards a resolution.
> Description of the Issue:
> Steps to Reproduce: curl -v -s -LOJ --trace-config all <url> 2>&1 | tee /tmp/curl-tar.log
> I've attached the log file for your reference: curl-tar.log
> I've also tried download it from the browser, but the issue persists. (maybe this proves the issue is on the server-side, not in curl)
> What I'm Hoping For:
> I would be incredibly appreciative if someone could:
>     • Analyze the log file and provide insights into the underlying cause of the error.
>     • Offer guidance on how to effectively resolve the issue.
>     • Share any relevant knowledge or resources that could aid in my understanding.
> I understand that debugging can be a complex process, and I sincerely value your time and expertise. Any assistance you can provide would be greatly welcomed.
> Thank you in advance for your consideration and support.

The command and url work for me. I can download the file. Looking at your log, the server sends a RST (reset) after a while, terminating the download intentionally. I cannot see why the server decides to do that. The download in the log is rather slow, maybe it has a protection against that. But that is pure speculation.

Kind Regards,
Stefan

**Reply by realyukii** [reply]
Thanks for the explanation! That makes things much clearer.
my internet connection might have been sluggish at the time. I encountered a similar error message while attempting a `git clone` of ArchWiki:
```
RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8): error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8 error: 2519 bytes of body are still expected fetch-pack: unexpected disconnect while reading sideband packet fatal: early EOF fatal: fetch-pack: invalid index-pack output)
```
Interestingly, Git offered a solution with the http.postBuffer option. It seems Git might leverage curl internally, and in this case, Git's built-in mechanisms handled the slower connection more gracefully.

Does curl itself have options specifically designed to address slow connections? (Since git seem to utilize curl, I suspect the cause might be similar) Knowing this would be incredibly helpful for others facing similar slow-connectivity woes in the future.
